### ISSUE TITLE
Calendar: Add onAfterSelect callback

### ISSUE THREAD

**Issue by PhilippGrashoff** [fomantic/Fomantic-UI#1518]
# Feature Request
Currently, there is the onSelect Callback. This callback is called BEFORE the selected date is transferred into the input field.
It would be nice to have a callback which is called AFTER the selected date was transferred to the input field.
My use case is simple: I implemented an auto-Submit form which submits whenever there is a **sensible** change in an input. Especially for a date, not every input is sensible, e.g. if someone types a date by keyboard. Here, submitting is only sensible when the enter key is pressed.
At the moment, I cant simply call the form submit in onSelect, as the value was not transferred to the input field at that point yet. Hence it is not sent to the server. A onAfterSelect callback would fix this.

At the moment I try to work around this in onSelect by something like:
```
                    function(date, mode) {
                        if(!date) {
                            return;
                        }
                       //set date to input before submitting the form
                        $(this).find('input').val(dateTodmY(date));
                        $("#someForm").form('submit');
                    }
```

**Comment by PhilippGrashoff** [comment]
But I do not want to disable the manual date input. Its very handy in case you have to enter the date which is for Example 3 Years in the future. Would need 36 clicks in the calendar.

**Comment by lubber-de** [comment]
Regarding the 36 clicks, its not necessary as i mentioned here https://github.com/fomantic/Fomantic-UI/issues/1517#issuecomment-642484258
We may put those two issues together into one to avoid postig the same ?

I still believe that additional suggested "AfterSelection" Event is not necessary because onChange and onBeforeChange are called right after onSelect is.
So, according to the usecase to do something _after_ the input field has been updated, you go with `onChange`

I understand, it's tricky tricky to find out when the user has actually stopped typing when onChange triggers, but this is part of your other issue #1517 

I would not even know where to call such an additional "afterSelect" in the code ( after onChange?  after onBeforeChange? after onSelect (as said, the input field is not updated at that time)

**Comment by PhilippGrashoff** [comment]
Could we pass an additional parameter to onChange like source which would e.g. "keyboard" or "calendar" (when someone clicks a date)? This would make working in onChange easier.
I still think the most important part is that incomplete dates are not guessed and transformed into actual dates.

**Comment by lubber-de** [comment]
@PhilippGrashoff Possibly 😉 
But you can already figure this out by checking for the `onSelect` event. It it was called right before the `onChange` you know ithe value comes from a cell selection. Otherwise it was a pure keyboard entry and you should validate the date value against your desired format.
See https://jsfiddle.net/lubber/18gwompk/11/

**Comment by PhilippGrashoff** [comment]
Yes, that is a nice idea setting some property using onSelect handler. Now only the wild date guessing needs to be fixed :) And thats issue #1517
